app/services/calculate_subject_specialisms.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+class CalculateSubjectSpecialisms
+  include ServicePattern
+
+  MAX_SUBJECTS_ALLOWED = 3
+
+  def initialize(course:)
```

Comment:
Does this need a course record? Could it not just receive the subject names directly? The course isn't used beyond the initializer

Reply:
Yeah, we could just pass in the course subjects.